controller/src/main/java/io/pravega/controller/server/ControllerService.java

```diff
@@ -129,6 +130,15 @@ public ControllerService(StreamMetadataStore streamStore, StreamMetadataTasks st
                   }, executor);
     }
 
+    public CompletableFuture<Boolean> checkStreamCreated(final String scope,
```

Comment:
We are also going to want to have `checkStreamWritable` etc. I think the most general way to do this would be to have an API that sends the State back to the client. (Maybe simplified). Then the client can check that the stream is writable by checking that it is not `unknown` or `sealed`

Additionally for writers, I'm not sure we should fail-fast on `creating`. That is sortof a valid state, in that we can delay until it is complete transparently to the application.

Reply:
thank you Tom for the comments; 
the previous impl is to reuse the `checkStreamExists` which already provided in controller; after discussion I think that: a stream is fully created indicated by stream stat is not UNKNOWN, CREATING  (list enum below);  `checkStreamExists` is insufficient for such verification; so the question is: " at what stream state(s), client can continue with the setup process" ?  the process contains the following operation: 
```
ConditionalOutputStream cond = condFactory.createConditionalOutputStream(segment, delegationToken, config.getEventWriterConfig());
SegmentMetadataClient meta = metaFactory.createSegmentMetadataClient(segment, delegationToken);
return new RevisionedStreamClientImpl<>(segment, in, out, cond, meta, serializer);
```

all the states:
```
public enum State {
    UNKNOWN,
    CREATING,
    ACTIVE,
    UPDATING,
    SCALING,
    COMMITTING_TXN,
    TRUNCATING,
    SEALING,
    SEALED;
  ...
```

Reply:
What @tkaitchuck  is suggesting is to make available additional information about the stream apart from just whether it was created successfully or not. 
This will help us determine if the stream is sealed or open. 

Instead of exposing all the internal states of a stream to the client, my suggestion is as follows:
have two separate apis - `checkStreamExists` and `checkStreamIsWritable`. 
Both of these apis return boolean. And their implementation on controller simply fetches the state and based on the state determines if the stream is created and not sealed.

Reply:
thank you @shivesh @Tom i'm going to remove the current API which returns Boolean value, instead, a new API will return (limited) stream state  - CREATING/ACTIVE/SEALING/SEALED/UNKNOWN,  API caller can then decide to continue/not create the client based on the returning state.

-- the above is based on discussion in slack; please comment for any further concerns.

Reply:
`getStreamState` added to replace previous `checkStreamExists`, so from client side we can get the stream state in detail.   "StreamState" has been added to describe the stream state.